File: elo_calculation.js
document.addEventListener('DOMContentLoaded', function() {
    document.querySelector('#calculateButton').onclick = calculate
});

function getExpectedScore(ratingDifference){
    return 1/(1+Math.pow(10, ratingDifference/400));
}

function calculate() {
    const firstRating = document.querySelector('#firstRating').value;
    const secondRating = document.querySelector('#secondRating').value;
    const k = document.querySelector('#k').value;
    const resultSelector = document.getElementsByName('result');
    for (var i = 0, length = resultSelector.length; i < length; i++) {
        if (resultSelector[i].checked) {
            result = Number(resultSelector[i].value);
            break;
        }
    }
    let ratingDifference = secondRating-firstRating;
    let expectedScore1 = getExpectedScore(ratingDifference);
    let expectedScore2 = getExpectedScore(-ratingDifference);
    let newRating1 = Math.round(Number(firstRating) + k*(result-expectedScore1));
    let newRating2 = Math.round(Number(secondRating) + k*(1-result-expectedScore2));
    alert(`Result is ${result}. New rating for the first player is ${newRating1}, for the second player is ${newRating2}`);
}

File: elo_calculation.py
def get_expected_score(rating_difference):
    return 1/(1+10**(rating_difference/400))

def calculate_new_elo(first_rating, second_rating, result, k):
    rating_difference = second_rating-first_rating
    expected_score_1 = get_expected_score(rating_difference)
    expected_score_2 = get_expected_score(-rating_difference)
    new_first_rating = round(first_rating + k*(result-expected_score_1))
    new_second_rating = round(second_rating + k*(1-result-expected_score_2))
    return new_first_rating, new_second_rating

